scripts/settlement/register_sequencer_to_hub.sh

```diff
@@ -1,19 +1,112 @@
+    exit 1
+  fi
+fi
+
+if [ "$SEQUENCER_KEY_PATH" = "" ]; then
+  DEFAULT_SEQUENCER_KEY_PATH="${ROLLAPP_HOME_DIR}/sequencer_keys"
+  echo "SEQUENCER_KEY_PATH is not set, using '${DEFAULT_SEQUENCER_KEY_PATH}'"
+  SEQUENCER_KEY_PATH=$DEFAULT_SEQUENCER_KEY_PATH
+fi
+
+if [ "$SEQUENCER_KEY_NAME" = "" ]; then
+  DEFAULT_SEQUENCER_KEY_NAME="sequencer"
+  echo "SEQUENCER_KEY_PATH is not set, using '${DEFAULT_SEQUENCER_KEY_PATH}'"
+  SEQUENCER_KEY_NAME=$DEFAULT_SEQUENCER_KEY_NAME
+fi
 
 #Register Sequencer
-DESCRIPTION="{\"Moniker\":\"${ROLLAPP_CHAIN_ID}-sequencer\",\"Identity\":\"\",\"Website\":\"\",\"SecurityContact\":\"\",\"Details\":\"\"}"
-SEQ_PUB_KEY="$("$EXECUTABLE" dymint show-sequencer)"
-BOND_AMOUNT="$(dymd q sequencer params -o json --node "$HUB_RPC_URL" | jq -r '.params.min_bond.amount')$(dymd q sequencer params -o json --node "$HUB_RPC_URL" | jq -r '.params.min_bond.denom')"
+# DESCRIPTION="{\"Moniker\":\"${ROLLAPP_CHAIN_ID}-sequencer\",\"Identity\":\"\",\"Website\":\"\",\"SecurityContact\":\"\",\"Details\":\"\"}"
+SEQ_PUB_KEY="$("$ROLLAPP_EXECUTABLE" dymint show-sequencer)"
+BOND_AMOUNT="$("$SETTLEMENT_EXECUTABLE" q sequencer params -o json --node "$HUB_RPC_URL" | jq -r '.params.min_bond.amount')$("$SETTLEMENT_EXECUTABLE" q sequencer params -o json --node "$HUB_RPC_URL" | jq -r '.params.min_bond.denom')"
+
+echo "$BOND_AMOUNT"
+
+if [ "$METADATA_PATH" = "" ]; then
+  DEFAULT_METADATA_PATH="${ROLLAPP_HOME_DIR}/init/sequencer-metadata.json"
+  echo "METADATA_PATH is not set, using '$DEFAULT_METADATA_PATH"
+  METADATA_PATH=$DEFAULT_METADATA_PATH
+
+  if [ ! -f "$METADATA_PATH" ]; then
+    echo "${METADATA_PATH} does not exist, would you like to use a dummy metadata file? (y/n)"
+    read -r answer
+
+    if [ "$answer" != "${answer#[Yy]}" ]; then
+      cat <<EOF > "$METADATA_PATH"
+{
+  "moniker": "Sample Moniker",
+  "details": "Some details about the sequencer",
+  "p2p_seeds": [
+    "seed1.example.com:26656",
+    "seed2.example.com:26656"
+  ],
+  "rpcs": [
+    "http://rpc1.example.com:26657",
+    "http://rpc2.example.com:26657"
+  ],
+  "evm_rpcs": [
+    "http://evm-rpc1.example.com:8545",
+    "http://evm-rpc2.example.com:8545"
+  ],
+  "rest_api_url": "http://restapi.example.com",
+  "explorer_url": "http://explorer.example.com",
+  "genesis_urls": [
+    "http://genesis1.example.com",
+    "http://genesis2.example.com"
+  ],
+  "contact_details": {
+    "website": "http://website.example.com",
+    "telegram": "https://t.me/example",
+    "x": "https://twitter.com/example"
+  },
+  "extra_data": "RXh0cmEgZGF0YSBzYW1wbGU=",  // Base64 encoded data
+  "snapshots": [
+    {
+      "snapshot_url": "http://snapshot1.example.com",
+      "height": 123456,
+      "checksum": "d41d8cd98f00b204e9800998ecf8427e"
+    },
+    {
+      "snapshot_url": "http://snapshot2.example.com",
+      "height": 789012,
+      "checksum": "e2fc714c4727ee9395f324cd2e7f331f"
+    }
+  ],
+  "gas_price": "1000000"
+}
```

Comment:
maybe it's better to store these in a file

Reply:
yeah, I though the same but that the user needs to run the script from a specific location and the json file needs to be at an expected location all the time... this way user can run the scripts from wherever he wants, that's why it's inline

since the sequencer metadata is optional, the dummy file will just be an empty json object, that's shorter